tests/smoke-tests/test-staking-rewards.ts

```diff
@@ -6,23 +6,307 @@ import { ApiPromise } from "@polkadot/api";
+          {
+            args: [_, accountId],
+          },
+          { bond, total, delegations },
+        ] = coll;
+        const candidateInfo = (
+          await limiter.schedule(() =>
+            predecessorApiAt.query.parachainStaking.candidateInfo(accountId as AccountId20)
+          )
+        ).unwrap();
+
+        const bondsMatch: boolean = bond.eq(candidateInfo.bond);
+        const delegationsTotalMatch: boolean =
+          delegations.length ==
+          Math.min(
+            candidateInfo.delegationCount.toNumber(),
+            predecessorApiAt.consts.parachainStaking.maxTopDelegationsPerCandidate.toNumber()
+          );
+        const totalSum: boolean = delegations
+          .reduce((acc: BN, curr) => {
+            return acc.add(curr.amount);
+          }, new BN(0))
+          .add(bond)
+          .eq(total);
+        return { collator: accountId.toString(), bondsMatch, delegationsTotalMatch, totalSum };
+      });
+
+      return Promise.all(allTasks);
+    });
+
+    const failures = results.filter((item) => Object.values(item).includes(false));
+    expect(
+      failures,
+      `Checks failed for collators: ${failures.map((a) => a.collator).join(", ")}`
+    ).to.be.empty;
+  });
+
+  it("should snapshot candidate delegation amounts correctly.", async function () {
+    // This test is so slow due to rate limiting, this should be off until a better solution appears
+    if (process.env.RUN_ATSTAKE_CONSISTENCY_TESTS != "true") {
+      debug("Explicit RUN_ATSTAKE_CONSISTENCY_TESTS flag not set to 'true', skipping test");
+      this.skip();
+    }
+
+    this.timeout(TWO_HOURS);
+    this.slow(TWO_HOURS);
+    const limiter = new Bottleneck({
+      maxConcurrent: 10,
+      minTime: 100,
+    });
+
+    // Function to check a single Delegator's delegation to a collator
+    const checkDelegatorDelegation = async (accountId: AccountId20, delegatorSnapshot) => {
+      const { delegations: delegatorDelegations }: PalletParachainStakingDelegator = (
+        (await limiter.schedule(() =>
+          predecessorApiAt.query.parachainStaking.delegatorState(delegatorSnapshot.owner)
+        )) as any
+      ).unwrap();
+
+      const delegationAmount = delegatorDelegations.find(
+        (candidate) => candidate.owner.toString() == accountId.toString()
+      ).amount;
+
+      // Querying for pending withdrawals which affect the total
+      const scheduledRequest = (
+        (await limiter.schedule(() =>
+          predecessorApiAt.query.parachainStaking.delegationScheduledRequests(
+            accountId as AccountId20
+          )
+        )) as any
+      ).find((a) => {
+        return a.delegator.toString() == delegatorSnapshot.owner.toString();
+      });
+
+      const expected =
+        scheduledRequest === undefined
+          ? delegationAmount
+          : delegationAmount.sub(
+              new BN((Object.values(JSON.parse(scheduledRequest.action))[0] as any).slice(3), "hex")
+            );
```

Comment:
This should probably be adapted to be used with types (also for readability) and to avoid future bugs if another action is introduced

```
if (scheduledRequest.isBondLess || scheduledRequest.isRevoke) {
   // sub
}
```

You'd also have access to the amount via `.asRevoke.toBN()`, etc.

Reply:
Done